File: mqtt-manager/src/main/java/com/ns/greg/library/mqtt_manager/external/MqttTopic.java
package com.ns.greg.library.mqtt_manager.external;

/**
 * @author Gregory
 * @since 2017/11/13
 */

public abstract class MqttTopic {

  private final String mqttTopic;

  public MqttTopic(String mqttTopic) {
    this.mqttTopic = mqttTopic;
  }

  @Override public String toString() {
    return "Topic: " + mqttTopic;
  }

  @Override public boolean equals(Object obj) {
    if (obj == this) {
      return true;
    }

    if (!(obj instanceof MqttTopic)) {
      return false;
    }

    MqttTopic target = (MqttTopic) obj;
    return mqttTopic.equals(target.mqttTopic);
  }

  public String getMqttTopic() {
    return mqttTopic;
  }
}

File: mqtt-manager/src/main/kotlin/com/ns/greg/library/mqtt_manager/external/topic/MqttTopic.kt
package com.ns.greg.library.mqtt_manager.external.topic

/**
 * @author gregho
 * @since 2019-06-19
 */
open class MqttTopic @JvmOverloads constructor(
  val topic: String = ""
) {

  override fun equals(other: Any?): Boolean {
    if (other == this) {
      return true
    }

    if (other !is MqttTopic) {
      return false
    }

    return topic == other.topic
  }

  override fun hashCode(): Int {
    return topic.hashCode()
  }

  override fun toString(): String {
    return topic
  }
}